### PR TITLE
Refactor the second storage so that we can support multi persist storages

### PR DESCRIPTION
**Pull request by git-hulk**

### CODE REVIEW

**Comment by git-hulk** [comment]
@iwinux @Napoleon0621 Can have a look, I won't add more test cases to make codecov happy since it's meaningless to cover those uncover codes.

File: storage/persistence/spanner/setup_test.go [hunk 1]
```diff
@@ -83,30 +67,27 @@ func CreateDatabase(ctx context.Context, uri string) error {
 		return nil
 	}
 
+	ddlBytes, err := ioutil.ReadFile("../../../scripts/schemas/spanner/ddls.sql")
+	if err != nil {
+		return fmt.Errorf("read ddls file: %w", err)
+	}
+	ddls := make([]string, 0)
+	for _, ddl := range strings.Split(string(ddlBytes), ";") {
```
Comment (Napoleon0621) [inline storage/persistence/spanner/setup_test.go:75]: There is only one table to be created, line 75 seems to be ineffective.

Comment (git-hulk) [reply storage/persistence/spanner/setup_test.go:75]: Not actually, at least we need to add some indexes like `pool_name + ready_time` to speedup the query. Besides, I think it's ok to ineffective in test case, it harms nothing. How do you think?

Comment (Napoleon0621) [reply storage/persistence/spanner/setup_test.go:75]: Yes, I agree.

**Review by Napoleon0621 — APPROVED** [review]

**Pull request merged**